src/scmrepo/git/backend/pygit2/__init__.py

```diff
@@ -646,13 +646,13 @@ def _get_remote(self, url: str) -> Generator["Remote", None, None]:
         except KeyError as exc:
             raise SCMError(f"'{url}' is not a valid Git remote or URL") from exc
 
-        parsed = urlparse(url)
-        if parsed.scheme in ("git", "git+ssh", "ssh") or url.startswith("git@"):
-            raise NotImplementedError
         if os.name == "nt" and url.startswith("file://"):
             url = url[len("file://") :]
-
-        yield self.repo.remotes.create_anonymous(url)
+        remote = self.repo.remotes.create_anonymous(url)
```

Comment:
`create_anonymous` will apply any configured URL rewrite rules (like `insteadOf`) before setting the returned `remote.url`, we should be checking the rewritten URL for SSH rather than the raw URL